### ISSUE TITLE
Add compile-time benchmark

### ISSUE THREAD

**Issue by noahgibbs** [Shopify/yjit-bench#114]
We'd like to use 30k_ifelse or similar as a compile-time benchmark. The right way to do it is probably to run with --yjit-call-threshold=1 for 1 iteration. But doing that doesn't fit well into our current harness structure. We'd like to be able to pass back arbitrary data, not just runs.

So this would require a bit of harness work for both yjit-bench and yjit-metrics, but then we could use our existing 30k_ifelse benchmark as a nice compile-time benchmark and track it.

**Comment by maximecb** [comment]
I think we could have a simple script that runs just the compile time benchmark and does nothing else, just running 30k_ifelse is probably sufficient. Though I suppose we could run a single iteration of railsbench as well, since we look at that benchmark a lot and it's more representative of a real workload.

**Comment by noahgibbs** [comment]
The only difficulty with Railsbench is that it's a lot harder to control the number of times each chunk of code is hit. The 30k benchmarks are so simple that you basically don't have any repetition. You could still run it, obviously. But the measurement would be likely to be less accurate.

**Comment by maximecb** [comment]
Yes that is true. Nevermind railsbench for now, we can just have the simplest possible script to estimate compilation time on 30k_ifelse.

Another idea: if we know how many ISEQs and blocks the benchmark compiles, then we can estimate compile time per ISEQ and per block.
